### PR TITLE
ESPnet Recipe for ASR on the Makerere Radio Speech Corpus

### PR DESCRIPTION
**Pull request by satvik-dixit**
## What?

This is a new recipe for preparing the Makerere Radio Speech Corpus dataset and training automatic speech recognition models on it.

## Details about the dataset:

The Makerere Radio Speech Corpus is a dataset in the Luganda language. It includes 20 hours of human-transcribed radio speech.

## Related Links

Details about the dataset: https://zenodo.org/records/5855017 
Paper on the dataset: https://arxiv.org/abs/2206.09790

### CODE REVIEW

**Comment by satvik-dixit** [comment]
The error ` ImportError: cannot import name 'kaiser' from 'scipy.signal' (/home/runner/work/espnet/espnet/tools/venv/lib/python3.10/site-packages/scipy/signal/__init__.py)` seems to be happening because `kaiser` should be imported from `scipy.signal.windows` according to the latest `scipy` version

**Comment by sw005320** [comment]
@jctian98, can you review this PR?

**Comment by jctian98** [comment]
@satvik-dixit Thanks for making this PR. I've left some comments.
Additionally, 
(1) You split the train/valid/test sets by yourself. Could you confirm that the official release of this dataset doesn't have that split? If they already have that split, we'd better follow it.
(2) You split the train_dev/test in the Python script; and then train/dev in the shell script. Could you put this logic together? either in Python or Shell should be fine.
(3) Seems the current recipe cannot pass the CI test due to the `scipy` version issue. After that issue is solved, please make sure to pass other CI tests.

**Comment by sw005320** [comment]
> @satvik-dixit Thanks for making this PR. I've left some comments above. Additionally, 
> (1) You split the train/valid/test sets by yourself. Could you confirm that the official release of this dataset doesn't have that split? If they already have that split, we'd better follow it. 
> (2) You split the train_dev/test in the Python script; and then train/dev in the shell script. Could you put this logic together? either in Python or Shell should be fine. 

@satvik-dixit, please respond to them.

> (3) Seems the current recipe cannot pass the CI test due to the `scipy` version issue. After that issue is solved, please make sure to pass other CI tests.

This is fixed.

**Comment by satvik-dixit** [comment]
1) That's right, the official release of this dataset doesn't have that split. Here are more about the dataset for reference:
Details about the dataset: https://zenodo.org/records/5855017
Paper on the dataset: https://arxiv.org/abs/2206.09790
2) Yes, I can include all of it in the Python script (data_prep.py)
3) Thanks Shinji

File: egs2/makerere/asr1/cmd.sh [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+# If jobs failed, your configuration might be wrong for your environment.
+#
+#
+# The official documentation for run.pl, queue.pl, slurm.pl, and ssh.pl:
+#   "Parallelization in Kaldi": http://kaldi-asr.org/doc/queue.html
+# =========================================================~
+
+
+# Select the backend used by run.sh from "local", "stdout", "sge", "slurm", or "ssh"
+cmd_backend='slurm'
```
Comment (sw005320) [inline egs2/makerere/asr1/cmd.sh:31]: Please do not change some environment related configs.
Please submit them as they are (e.g., `cmd.sh` and `slurm.conf`)

File: egs2/makerere/asr1/README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
+<!-- Generated by scripts/utils/show_asr_result.sh -->
+# RESULTS
+## Environments
+- date: `Thu Mar 28 00:34:47 EDT 2024`
+- python version: `3.10.13 | packaged by conda-forge | (main, Dec 23 2023, 15:36:39) [GCC 12.3.0]`
+- espnet version: `espnet 202402`
+- pytorch version: `pytorch 2.0.1`
+- Git hash: `eed7751c910977290ef9a177ea0942a0e3c2fd35`
+  - Commit date: `Mon Mar 25 18:26:50 2024 +0000`
```
Comment (sw005320) [inline egs2/makerere/asr1/README.md:9]: Please upload the model to the HF hub and add a link here

**Comment by jctian98** [comment]
Hi @satvik-dixit , is there anything I can help?

**Comment by jctian98** [comment]
@satvik-dixit Thanks for the current progress. Shinji and I left some comments before. Please also kindly address those comments :)

File: egs2/makerere/asr1/cmd.sh (hunk 1 above)
Comment (satvik-dixit) [reply egs2/makerere/asr1/cmd.sh]: I have made the suggested changes in `cmd.sh` and `slurm.conf`

File: egs2/makerere/asr1/README.md (hunk 2 above)
Comment (satvik-dixit) [reply egs2/makerere/asr1/README.md:9]: I have uploaded the model to the HF hub. Here's the link: https://huggingface.co/satvik-dixit/asr_makerere.

**Comment by sw005320** [comment]
LGTM.
I'll merge it with some checks.

**Comment by sw005320** [comment]
Please add a corpus description to https://github.com/espnet/espnet/blob/master/egs2/README.md

**Comment by satvik-dixit** [comment]
Added the corpus description to readme (https://github.com/espnet/espnet/blob/master/egs2/README.md)

**Comment by sw005320** [comment]
Thanks a lot!

**Pull request merged**